transport/src/main/java/io/netty/channel/nio/NioHandler.java

```diff
@@ -617,14 +617,14 @@ private void processSelectedKey(SelectionKey k, AbstractNioChannel ch) {
             }
 
             // Process OP_WRITE first as we may be able to write some queued buffers and so free memory.
-            if ((readyOps & SelectionKey.OP_WRITE) != 0) {
+            if (k.isWritable()) {
                 // Call forceFlush which will also take care of clear the OP_WRITE once there is nothing left to write
                 ch.unsafe().forceFlush();
             }
 
             // Also check for readOps of 0 to workaround possible JDK bug which may otherwise lead
             // to a spin loop
-            if ((readyOps & (SelectionKey.OP_READ | SelectionKey.OP_ACCEPT)) != 0 || readyOps == 0) {
+            if (readyOps == 0 || k.isReadable() || k.isAcceptable()) {
```

Comment:
this will be a bit slower as you do one extra check compared to bitwise stuff

Reply:
@kezhenxu94 basically what I am saying that before you needed 2 checks while with your its 3 now. That said I think it really not matters too much.